corehq/apps/custom_data_fields/static/custom_data_fields/js/custom_data_fields.js

```diff
@@ -58,6 +58,8 @@ function CustomDataField () {
 function CustomDataFieldsModel () {
     var self = this;
     self.data_fields = ko.observableArray();
+    // The data field that the "remove field modal" currently refers to.
+    self.modalField = ko.observable();
```

Comment:
Since there was one modal per row before, the modal markup contained the context, but now we need to store it somewhere else.